src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -799,7 +799,7 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                             // only report unused usings if we have success.
                             if (success)
                             {
-                                compilation.ReportUnusedImports(null, diagnostics, cancellationToken);
+                                compilation.ReportAndFilterUnusedImports(diagnostics, cancellationToken);
```

Comment:
❓ Why is this call different from `ReportUnusedImports`? In other words, why did you not update `ReportUnusedImports` to call `FilterAndAppendDiagnostics`?

Reply:
I looked at the various other callers of `ReportUnusedImports` and saw that they are already called before the filtering. The two callers that I modified are the only two that lacked filtering.
Some of the tests I added cover those other scenarios.